hormonal-hedonists/TLOA/core/game.py

```diff
@@ -0,0 +1,160 @@
+        self._ship_destroyed: Sound = SoundLoader.load(AUDIO_PATH.format("ship_explosion.mp3"))
+
+        # Create & initialize Mesh instance for Incident Sun rays
+        self.sun_rays = LightRays(point=LIGHT_SOURCE_POS, surface=self.mirror.mirror_axis)
+
+        # Create & initialize Mesh instance for Reflected Sun rays/Death Rays
+        self.death_rays = LightRays(
+            point=Vector(MIRROR_CANNON_RANGE, LANE_BOUNDS[self.mirror.state][1]),
+            surface=self.mirror.mirror_axis,
+            angle=self.mirror.angle)
+
+        self.ship_lanes: List[List[BrownShip]] = [[] for _ in range(NUMBER_OF_LANES)]
+
+        self._event: Optional[ClockEvent] = None
+        self.register_event_type('on_add_ship')
+        self.register_event_type('on_remove_ship')
+        self.bind(health=self.on_health_change)
+
+    def start(self):
+        Logger.info('Start game')
+
+        self._sound: Sound = SoundLoader.load(AUDIO_PATH.format("island_background.wav"))
+        self._sound.loop = True
+        self._sound.play()
+
+        self.running = True
+
+        Clock.schedule_interval(self.step, TICK)
+        Clock.schedule_interval(lambda _: self.spawn_ship(), SHIP_SPAWN_RATE)
+
+    def step(self, dt):
+        if not self.running:
+            return False
+        elif self.pause_game:
+            return
+
+        # step ships and remove any dead ones
+        for lane in self.ship_lanes:
+            for ship in lane[:]:
+                if ship.is_dead:
+                    lane.remove(ship)
+                    self.dispatch('on_remove_ship', ship)
+                    Logger.info("%s died", ship)
+                    continue
+
+                ship.step(dt, self)
+
+        # Trace Sun rays onto the mirror
+        self.sun_rays.trace(point=LIGHT_SOURCE_POS, surface=self.mirror.mirror_axis)
+        self.sun_rays.target = False
+
+        # Track the closest ship in the active lane
+        if self.closest_ship:
+            # focus the Death Rays to the ship's X position.
+            death_rays_focus_x = self.closest_ship.shape.x + LIGHT_FOCUS_OFFSET.x
+            self.death_rays.target = True
+
+            # Deal damage to the ship.
+            if self.closest_ship.id == GoldenShip.id:
+                self.closest_ship.health -= 0.5    # Deal less Damage to Golden Ships
+
+            if self.closest_ship.id == BrownShip.id:
+                self.closest_ship.health -= 1.0    # Deal more Damage to Brown Ships.
+        else:
+            death_rays_focus_x = MIRROR_CANNON_RANGE
+            self.death_rays.target = False
+
+        # Trace Death rays onto the closest ship in the active lane.
+        self.death_rays.trace(
+            point=Vector(death_rays_focus_x,
+                         LIGHT_FOCUS_OFFSET.y+LANE_BOUNDS[self.mirror.state][1]),
+            surface=self.mirror.mirror_axis)
+
+    # This property returns the closest ship in the active lane.
```

Comment:
This pseudo-docstring should have a been a real docstring since docstrings have a special meaning in Python/in the Python data model. The contents of a docstring will be bound to the object's `__doc__` attribute and be used things like auto-generated help information for the object (e.g., `help(Game.closest_ship)` within Python).

You can read more about this in [PEP 257 -- Docstring Conventions](https://www.python.org/dev/peps/pep-0257/) and the [Data Model page](https://docs.python.org/3/reference/datamodel.html) in the documentation (search for `__doc__` on that page).